### PR TITLE
Adds a folder for torch.aten.broadcast_to operation.

### PR DESCRIPTION
**Pull request by ttjost**

### CODE REVIEW

File: python/torch_mlir_e2e_test/test_suite/basic.py [hunk 1]
```diff
@@ -1330,8 +1330,32 @@ def forward(self, x, y):
...
+        return torch.ops.aten.sub(x, y)
+
+
+@register_test_case(module_factory=lambda: BroadcastListConstructWithMinusOneModule())
+def BroadcastListConstructWithMinusOneModule_basic(module, tu: TestUtils):
+    module.forward(tu.rand(3, 1, 8), tu.rand(3, 1, 8))
+
+# ==============================================================================
+
+
```
Comment (flemairen6) [inline python/torch_mlir_e2e_test/test_suite/basic.py:1358]: Bunch of unnecessary changes there and before the test

File: lib/Dialect/Torch/IR/TorchOps.cpp [hunk 2]
```diff
@@ -1175,6 +1175,37 @@ void AtenSizeOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
...
+
+  // May fold if any of the dims == -1, which means its value should be preserved.
+  auto isMinusOne = [&](int64_t &dimension) { return dimension == -1; };
+  if (llvm::none_of(resultShape, isMinusOne))
+    return nullptr;
+
+  auto selfType = this->getSelf().getType().dyn_cast<ValueTensorType>();
+  if (!selfType || !selfType.hasSizes())
+    return nullptr;
+
```
Comment (mgehre-amd) [inline lib/Dialect/Torch/IR/TorchOps.cpp:1196]: Ah, you are removing the whole BroadcastToOp.

I was actually thinking to just update existing resultshape from -1 to the actual shape of the input in that dimension.
So if some broadcasting needs to happen, it can still happen. But we are guaranteed to not see a `-1` in the resultShape.
Now I realize that the resultShape are not an attribute on the op, but a different op, so this kind of thing cannot happen in the the ::fold method.

Anyway, does this change fix all the affected models? If yes, then its good enough and we don't need to do my more complicated idea.

**Review by mgehre-amd — APPROVED** [review]

**Review by flemairen6 — APPROVED** [review]

**Comment by ttjost** [comment]
@mgehre-amd , do you mind checking the PR again, please? I actually remove the `fold` function and added that test for `-1` as part of the legalization. Thanks

**Review by mgehre-amd — APPROVED** [review]

**Comment by mgehre-amd** [comment]
CI failed because your new test case passes in stablehlo. Please add your test to the stablehlo pass list in xfail_sets.py
